settings.d/shellhooks.yml.example

```diff
@@ -1,5 +1,6 @@
 ---
-# always enable HTTPS only
+# Use of HTTPS only is strongly advised. Exposing shellhooks over http is
+# technically possible but allows anyone to run things on your proxy.
```

Comment:
If you really want to get technical, this isn't quite right. Today there is no check on who does connect anyway so HTTP or HTTPS doesn't matter (@evgeni has mentioned this on foreman-security@).

If you add `authorize_with_ssl_client` in [shellhooks_api.rb](https://github.com/theforeman/smart_proxy_shellhooks/blob/master/lib/smart_proxy_shellhooks/shellhooks_api.rb) then the SSL certificates will be verified. This doesn't make a difference when HTTP is used.

If you add `authorize_with_trusted_hosts` the connecting host will be verified. On HTTPS this uses the client certificate if present. Otherwise it falls back to reverse DNS.

Reply:
The idea I wanted to convey was "you should use https. you can use http, but you should know what you're doing if you do". I definitely did not want to fix security issues here. Any suggestions?

Reply:
Today this recommendation isn't true: anyone can run anything on the proxy if this plugin is enabled and from the server side there's no security benefit in using https without actually checking it. Stating there is a benefit in the example is misleading.

Reply:
How about this, since HTTP is so dangerous and irrelevant, what if we completely disable it for this plugin? We simply wont start HTTP listener and drop a comment here that only HTTPS is supported.

That authorization problem is a different conversation, patch incoming :-)

Reply:
> If you really want to get technical, this isn't quite right. Today there is no check on who does connect anyway so HTTP or HTTPS doesn't matter (@evgeni has mentioned this on foreman-security@).
> 
> If you add `authorize_with_ssl_client` in [shellhooks_api.rb](https://github.com/theforeman/smart_proxy_shellhooks/blob/master/lib/smart_proxy_shellhooks/shellhooks_api.rb) then the SSL certificates will be verified. This doesn't make a difference when HTTP is used.

This method should really be called `auth_any_ssl_client` as it doesn't check more than the CA.

> 
> If you add `authorize_with_trusted_hosts` the connecting host will be verified. On HTTPS this uses the client certificate if present. Otherwise it falls back to reverse DNS.
> 
> That means you really want to add both. See https://github.com/theforeman/smart-proxy/blob/fdeef1dc6febcfae22c8d3273cb18d6bdeb31a23/modules/dhcp/dhcp_api.rb#L5-L6

I think trusted only is more correct here.

Reply:
How about: https://github.com/theforeman/smart_proxy_shellhooks/pull/2

Reply:
So my proposal to completely remove HTTP endpoint (but keep it via an environmental variable) was not liked. I closed the PR.

What do you really suggest? Apologies for the delay but I would really like to unblock the installer patches for the webhooks and the shellhook plugins.

Reply:
I think we might be running into just different expectations.

In this PR, I would like to fix the comment.

Then there is a security-related concern that is currently being investigated (we are waiting for confirmation from the security-response team). I plan to file a PR shortly, that is a bug, confirmed, fix is known.

If you prefer both to be solved in a single PR I can do that as well.

Reply:
How about this: https://github.com/theforeman/smart_proxy_shellhooks/pull/5